examples/Identifying Local Environments in a Complex Crystal.ipynb

```diff
@@ -0,0 +1,445 @@
+      "text/plain": [
+       "<fresnel.util.ImageArray at 0x12aff8910>"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import fresnel\n",
+    "\n",
+    "# Color particles by different local environments identified\n",
+    "colors = np.empty((snapshot.particles.N, env_cluster.num_clusters))\n",
+    "colors[typeid == 0] = fresnel.color.linear([0.99, 0, 0])\n",
+    "colors[typeid == 1] = fresnel.color.linear([0.99, 0.99, 0])\n",
+    "colors[typeid == 2] = fresnel.color.linear([0, 0.93, 0.95])\n",
```

Comment:
You can do something that scales more easily to any number of clusters by using NumPy indexing:
```python
type_colors = np.array([
    fresnel.color.linear([0.99, 0, 0]),
    fresnel.color.linear([0.99, 0.99, 0]),
    fresnel.color.linear([0, 0.93, 0.95]),
])
colors = type_colors[typeid]  # Indexes the (3, 3) array using (N,) types to make an (N, 3) array

Reply:
This is cool. I had no idea we could do something like this.